### PR TITLE
PROD-419/share-reveal-single

### PR DESCRIPTION
**Pull request by danijel-radomiljac**

### CODE REVIEW

File: app/api/og/share-claim-all/route.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+          </div>
+          <div tw="flex w-[10px]"></div>
+          <div tw="flex flex-col grow">
+            <ChompyGuitarIcon width={220} height={200} />
+          </div>
+        </div>
+      </div>
+    ),
+    {
+      width: 1200,
```
Comment (marvinmarnold) [inline app/api/og/share-claim-all/route.tsx:127]: Should be same size as designs.

File: app/a/[txHash]/page.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    },
+  };
+}
+
+export default async function Page() {
+  const headersList = headers();
+  const referer = headersList.get("referer");
+
+  redirect(
+    `/application?${referer === "x" ? "utm_source=x&utm_medium=social&" : ""}utm_campaign=app_share&utm_content=claim_all`,
```
Comment (marvinmarnold) [inline app/a/[txHash]/page.tsx:27]: Pleas see issue description for details about constructing the UTM params.

File: app/api/og/share-claim-all/route.tsx [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+import { decryptString } from "@/app/utils/crypto-js";
+import { ImageResponse } from "next/og";
+
+export async function GET(request: Request) {
+  const { searchParams } = new URL(request.url);
+  const encryptedHash = searchParams.get("encryptedTxHash");
+
+  if (!encryptedHash) return;
+
+  const txHash = decryptString(encryptedHash);
```
Comment (marvinmarnold) [inline app/api/og/share-claim-all/route.tsx:13]: We still need to solve the long URL issue

**Review by marvinmarnold — CHANGES_REQUESTED** [review]
I'll look at this in more detail after the claim all share flow gets over the line so we can make sure they follow the same patterns.

**Review by marvinmarnold — CHANGES_REQUESTED** [review]
Left feedback: https://linear.app/gator/issue/PROD-419/share-reveal-single#comment-144f3880

**Comment by marvinmarnold** [comment]
@danijel-radomiljac merge conflicts

**Comment by danijel-radomiljac** [comment]
> @danijel-radomiljac og image is not loading via link [opengraph.xyz/url/https%3A%2F%2Fchomp-qrh4u8kpn-gator-labs.vercel.app%2Fs%2F5zmDqmV4nz](https://www.opengraph.xyz/url/https%3A%2F%2Fchomp-qrh4u8kpn-gator-labs.vercel.app%2Fs%2F5zmDqmV4nz)

hmm can you try again, works fine for me: https://socialsharepreview.com/?url=https://chomp-git-prod-419-share-reveal-single-gator-labs.vercel.app/s/4xof7a3KQE&1541

**Comment by marvinmarnold** [comment]
This still isn't working for me.

> hmm can you try again, works fine for me: https://socialsharepreview.com/?url=https://chomp-git-prod-419-share-reveal-single-gator-labs.vercel.app/s/4xof7a3KQE&1541

Image not loading when I visit your socialsharepreview link
Screenshot 2024-11-27 at 5 58 24 PM

Also, I notice that the image doesn't load in Twitter when I follow the embed link. But it does work when I copy the URL and post on Twitter manually. Could this be because it looks like the manual copy link has something extra at the end? `https://chomp-git-prod-419-share-reveal-single-gator-labs.vercel.app/s/2YYdspnLkk&1552`, extra = `&1552`.
Screenshot 2024-11-27 at 5 59 02 PM

**Review by marvinmarnold — APPROVED** [review]

**Comment by Yihwan** [comment]
overall the flow looks good – it seems the image gets cut off though, wasn't sure if that's expected. 
Screenshot 2024-12-02 at 09 43 02

**Pull request merged**